AzureMonitorAgent/agent.py

```diff
@@ -1330,7 +1388,7 @@ def log_and_exit(operation, exit_code = 1, message = ''):
     """
     Log the exit message and perform the exit
     """
-    if exit_code is 0:
+    if exit_code == 0:
```

Comment:
I can't comment on it bc. there is no diff, but in the `is_vm_supported_for_extension` function there is an instance of `is not` between two ints, which should probably also take this change. Might be worth auditing for more instances of bad use of `is [not]` in the old functions taken from OMS

Reply:
This was the only instance in this file. Fixed now